src/lib/server/database/schema/app.ts

```diff
@@ -95,13 +95,32 @@ export const studentRank = app.table(
     userId: ulid('user_id')
       .notNull()
       .references(() => user.id, { onUpdate: 'cascade' }),
-    labs: text('labs').array().notNull(), // REFERENCES (EACH ELEMENT OF app.lab) app.lab (lab_id)
   },
   ({ draftId, userId }) => [primaryKey({ columns: [draftId, userId] })],
 );
 export type StudentRank = typeof studentRank.$inferSelect;
 export type NewStudentRank = typeof studentRank.$inferInsert;
 
+export const studentRankLab = app.table(
+  'student_rank_lab',
+  {
+    draftId: bigint('draft_id', { mode: 'bigint' })
+      .notNull()
+      .references(() => draft.id, { onUpdate: 'cascade' }),
+    userId: ulid('user_id')
+      .notNull()
+      .references(() => user.id, { onUpdate: 'cascade' }),
+    labId: text('lab_id')
+      .notNull()
+      .references(() => lab.id, { onUpdate: 'cascade' }),
+    index: bigint('index', { mode: 'bigint' }).notNull(),
+    remark: text('remark').notNull().default(''),
+  },
+  ({ draftId, userId, labId }) => [primaryKey({ columns: [draftId, userId, labId] })],
```

Comment:
Really love the decision to use `labId` over `index` as the primary key component, by the way. My initial brainstorm were to use `index`, but in hindsight, it is _really_ clever that you implicitly enforced `UNIQUE` constraints here by using `labId` instead. 👏